src/utils/convert_unit.ts

```diff
@@ -0,0 +1,26 @@
+import type { ByteUnitType } from '../types';
+
+const conversionRates = {
+	B: 0,
+	KB: 1,
+	MB: 2,
+	GB: 3
+};
+
+/**
+ *
+ * Converts byte count from a byte unit type into another byte unit type
+ * Works with bytes, kilobytes, megabytes, and gigabytes. Measured in binary
+ *
+ * @param count number of bytes in the specified input byte unit type
+ * @param inputUnit byte unit type for input count parameter
+ * @param outputUnit byte unit type to be converted into
+ * @returns byte count converted into the specified output unit
+ *
+ * @remarks Decimal count is not considered in this function
```

Comment:
Decimal values do actually work. They just might not make sense when there's a decimal input and the inputUnit is 'B'. We might want to throw an error for that...

Reply:
Consider that carefully though because unless we're disallowing fractional byte types in the UI, we'll end up throwing errors a lot.

Reply:
Actually no we won't, because there's no option for 'B' in the dropdown :)